src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -676,7 +673,8 @@ bool Node::RunConsensusOnMicroBlockWhenShardLeader() {
 
   if (m_mediator.m_ds->m_mode == DirectoryService::Mode::IDLE &&
       !m_mediator.GetIsVacuousEpoch()) {
-    std::this_thread::sleep_for(chrono::milliseconds(TX_DISTRIBUTE_TIME_IN_MS));
+    std::this_thread::sleep_for(chrono::milliseconds(TX_DISTRIBUTE_TIME_IN_MS +
```

Comment:
Why do we need to add the extra delay here for shard consensus?

Reply:
Ah I mean, shouldn't this be for both shard MB and DS MB consensus?

Reply:
the DS leader already have an extra sleep in RunConsensusOnFinalBlockWhenPrimary

Reply:
The purpose of this sleep is to make sure the backup already finished the txn processing before the announcement was received